src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +171,66 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  /**
+   * Add the given ttl at all the version of the document identified by
+   * the searchKey
+   */
+
+  public updateTTLForAllVersion(
```

Comment:
I'd rather add a method to patch any field of all versions (general purpose), if you really need it to be here.

Reply:
I do not agree to be honest, i believe that the `ttl` should be the unique property to be updated for all the versions of a document, updating other properties for all the versions would be against the concept of versioning imho ...

Reply:
In this case I wonder if we need this to be here. It looks like we need some data mangling that, I guess, will be run once. Or is this a common need that may arise regularly in the future?

Reply:
This is a common need that we have 👍🏽

Reply:
We should careful about this function because when it is applied to collections where documents have a lot of versions (aka profile, preferences) it could become very expensive on CPU/memory and RU consumption.